### PR TITLE
Waiver update

### PR DESCRIPTION
**Pull request by sclark**
Added digital signature and prevent showing the safety video to a user more than once

### CODE REVIEW

**Comment by sclark** [comment]
The digital signature is still pending approval from Andrew and Student Activities

File: app/controllers/waivers_controller.rb [hunk 1]
```diff
@@ -11,7 +11,9 @@ def new
     if @user.has_signed_waiver
       flash[:notice] = "You have already agreed to the release."
     end
-    
+
+    @should_see_video = !current_user.participant.is_scc? or params[:participant_id] == nil
```
Comment (pkoenig10) [inline app/controllers/waivers_controller.rb]: The user resolution logic already exists above in this method.  Should be `@should_see_video = @user.is_scc?`

File: app/views/waivers/new.html.erb [hunk 2]
```diff
@@ -1,108 +1,103 @@
...
-
-  I understand that there are risks (including, but not limited to, the risk of injury, disability and death)
-
-  associated with my participation in the Activities. I voluntarily assume all the risks. In consideration of
 
-  the benefits of participation in the Activities, I hereby, on behalf of myself and those acting on my
+<h2><%= Time.now.year %> SPRING CARNIVAL MIDWAY RELEASE OF LIABILITY</h2>
 
-  behalf, irrevocably and unconditionally release, waive and promise not to sue CMU from/for any and all
+<p>
```
Comment (pkoenig10) [inline app/views/waivers/new.html.erb]: In general, if you are going to refactor, I'd prefer you do it in a subsequent PR so its easier to tell what's changed and what hasn't when doing a review.

File: app/views/waivers/new.html.erb [hunk 3]
```diff
@@ -1,108 +1,103 @@
...
+        'onStateChange': onPlayerStateChange
+      }
+    });
   }
 
-  // autoplay video
+  // autoplay video if there were no flashed errors
   function onPlayerReady(event) {
-  event.target.playVideo();
+    <%= (@should_see_video and flash[:error] == nil)? "event.target.playVideo();" : "showWaiver();" %>
```
Comment (pkoenig10) [inline app/views/waivers/new.html.erb]: Change the controller logic to make `@should_see_video` false if there are any errors and remove the check here.

File: app/controllers/waivers_controller.rb (hunk 1 above)
Comment (sclark) [inline app/controllers/waivers_controller.rb]: `@user.is_scc?` should not check for `nil` participant, but I will change the first part

File: app/views/waivers/new.html.erb (hunk 2 above)
Comment (sclark) [inline app/views/waivers/new.html.erb]: This clean up had to do with changes in how the video is shown (`@should_show_video`) so it was included. Other refactoring will be done on new branches.

File: app/views/waivers/new.html.erb (hunk 3 above)
Comment (sclark) [inline app/views/waivers/new.html.erb]: I would but `flash[:error]` not necessarily `nil` in the controller `new` method

Comment (pkoenig10) [inline app/views/waivers/new.html.erb]: We redirect to the new route when an error occurs so we will recompute `@should_see_video`

File: app/controllers/waivers_controller.rb [hunk 4]
```diff
@@ -11,7 +11,9 @@ def new
     if @user.has_signed_waiver
       flash[:notice] = "You have already agreed to the release."
     end
-    
+
+    @should_see_video = !@user.is_scc? or params[:participant_id] == nil
```
Comment (pkoenig10) [inline app/controllers/waivers_controller.rb]: The nil check here is unnecessary

File: app/views/waivers/new.html.erb (hunk 3 above)
Comment (sclark) [inline app/views/waivers/new.html.erb]: When testing this I couldn't get the video to skip after an error when including the `flash[:error]` check in the controller

Comment (pkoenig10) [inline app/views/waivers/new.html.erb]: Alright no worries, we can always come back and clean this up another time.

**Pull request merged**